packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -116,13 +116,16 @@ class ChromeLauncher {
   /// port is picked automatically.
   ///
   /// `skipCheck` does not attempt to make a devtools connection before returning.
-  Future<Chrome> launch(String url, { bool headless = false, int debugPort, bool skipCheck = false, Directory cacheDir }) async {
+  ///
+  /// `userDataDir` is Chrome's user data dir. If null, a temporary directory
+  /// is created automatically.
+  Future<Chrome> launch(String url, { bool headless = false, int debugPort, bool skipCheck = false, Directory cacheDir, Directory userDataDir}) async {
     if (_currentCompleter.isCompleted) {
       throwToolExit('Only one instance of chrome can be started.');
     }
 
     final String chromeExecutable = findChromeExecutable(_platform, _fileSystem);
-    final Directory userDataDir = _fileSystem.systemTempDirectory.createTempSync('flutter_tools_chrome_device.');
+    userDataDir ??= _fileSystem.systemTempDirectory.createTempSync('flutter_tools_chrome_device.');
 
     if (cacheDir != null) {
```

Comment:
If you're specifying a user data dir I think the `_restoreUserSessionInformation` functionality needs to be disabled. This will copy some saved profile information from the project cache into the temporary user data dir. If you're intentionally re-using a data dir this may blow away some data unexpectedly

Reply:
I agree. I changed this behavior.